mayastor/tests/nexus_rebuild.rs

```diff
@@ -0,0 +1,144 @@
+use crossbeam::channel::unbounded;
+pub mod common;
+
+use mayastor::{
+    bdev::{nexus_create, nexus_lookup},
+    core::{mayastor_env_stop, MayastorCliArgs, MayastorEnvironment, Reactor},
+};
+
+static DISKNAME1: &str = "/tmp/disk1.img";
+static BDEVNAME1: &str = "aio:///tmp/disk1.img?blk_size=512";
+
+static DISKNAME2: &str = "/tmp/disk2.img";
+static BDEVNAME2: &str = "aio:///tmp/disk2.img?blk_size=512";
+
+static NEXUS_NAME: &str = "rebuild_test";
+static NEXUS_SIZE: u64 = 10 * 1024 * 1024; // 10MiB
+
+#[test]
+fn rebuild_test() {
+    common::delete_file(&[DISKNAME1.into(), DISKNAME2.into()]);
+    common::truncate_file(DISKNAME1, NEXUS_SIZE / 1024);
+    common::truncate_file(DISKNAME2, NEXUS_SIZE / 1024);
+
+    test_init!();
+
+    Reactor::block_on(rebuild_test_start());
+
+    common::delete_file(&[DISKNAME1.into(), DISKNAME2.into()]);
+}
+
+async fn rebuild_test_start() {
+    create_nexus().await;
+
+    let nexus = nexus_lookup(NEXUS_NAME).unwrap();
+    let device = nexus.share(None).await.unwrap();
+
+    let nexus_device = device.clone();
+    let (s, r) = unbounded::<String>();
+    std::thread::spawn(move || s.send(dd_urandom(&nexus_device)));
+    reactor_poll!(r);
+
+    let nexus_device = device.clone();
+    let (s, r) = unbounded::<String>();
+    std::thread::spawn(move || {
+        s.send(compare_nexus_device(&nexus_device, DISKNAME1, true))
+    });
+    reactor_poll!(r);
+
+    let nexus_device = device.clone();
+    let (s, r) = unbounded::<String>();
+    std::thread::spawn(move || {
+        s.send(compare_nexus_device(&nexus_device, DISKNAME2, false))
+    });
+    reactor_poll!(r);
+
+    // add the second child
+    nexus.add_child(BDEVNAME2).await.unwrap();
+
+    // kick's off the rebuild (NOWAIT)
+    nexus.start_rebuild(BDEVNAME2).await.unwrap();
+
+    // crude wait for the rebuild
+    let (s, r) = unbounded::<String>();
+    std::thread::spawn(move || {
+        s.send(delay(std::time::Duration::from_secs(1)))
+    });
+    reactor_poll!(r);
+
+    let (s, r) = unbounded::<String>();
+    std::thread::spawn(move || {
+        s.send(compare_devices(DISKNAME1, DISKNAME2, true))
+    });
+    reactor_poll!(r);
+
+    mayastor_env_stop(0);
+}
+
+fn delay(dur: std::time::Duration) -> String {
+    std::thread::sleep(dur);
+    "done".to_string()
+}
+
+async fn create_nexus() {
+    let ch = vec![BDEVNAME1.to_string()];
+    nexus_create(NEXUS_NAME, NEXUS_SIZE, None, &ch)
+        .await
+        .unwrap();
+}
+
+pub fn dd_urandom(device: &str) -> String {
```

Comment:
the helper functions should be moved into test/common.rs

Reply:
yeah I wondered about that, but didn't want to pollute it without approval, I'll do so now :)